components/org.wso2.carbon.identity.api.server.idp/org.wso2.carbon.identity.api.server.idp.v1/src/main/java/org/wso2/carbon/identity/api/server/idp/v1/core/ServerIdpManagementService.java

```diff
@@ -2772,4 +2776,125 @@ private static String includeData(Constants.ErrorMessage error, String data) {
         }
         return message;
     }
+
+    /**
+     * Validate the claim configs of an IDP.
+     *
+     * @param tenantDomain Tenant domain.
+     * @param claims       Claim configs.
+     * @throws IdentityProviderManagementException If an error while validating the claim configs or if an invalid
+     *                                             config is found.
+     */
+    private void validateClaims(String tenantDomain, Claims claims) throws IdentityProviderManagementException {
+
+        if (claims == null) {
+            return;
+        }
+        String userClaimURI = claims.getUserIdClaim() == null ? null : claims.getUserIdClaim().getUri();
+        String roleClaimURI = claims.getRoleClaim() == null ? null : claims.getRoleClaim().getUri();
+        List<org.wso2.carbon.identity.api.server.idp.v1.model.ClaimMapping> claimMappings = claims.getMappings();
+
+        // EMPTY claimMappings indicate that the IDP is using local claim dialect.
+        if (CollectionUtils.isEmpty(claimMappings)) {
+            List<LocalClaim> localClaimsList = getLocalClaimURIs(tenantDomain);
+            Set<String> claimURIs = localClaimsList.stream().map(LocalClaim::getClaimURI).collect(Collectors.toSet());
+            // Validate userClaimURI and roleClaimURI.
+            if (StringUtils.isNotBlank(userClaimURI) && claimURIs.contains(userClaimURI)) {
```

Comment:
Shoudn't this be not contains? 

!claimURIs.contains(userClaimURI)

Reply:
Fixed in f44b32e0efda15349b0bbf3d014763ee0b8297ad